src/d3a/models/area/markets.py

```diff
@@ -15,84 +15,175 @@
-        self.balancing_markets = OrderedDict()  # type: Dict[DateTime, BalancingMarket]
-        # Past markets
-        self.past_markets = OrderedDict()  # type: Dict[DateTime, Market]
-        self.past_balancing_markets = OrderedDict()  # type: Dict[DateTime, BalancingMarket]
-        self._indexed_future_markets = {}
-
-    @property
-    def indexed_future_markets(self):
-        return self._indexed_future_markets
-
-    @property
-    def all_spot_markets(self):
-        return list(self.markets.values()) + list(self.past_markets.values())
-
-    @property
-    def all_future_spot_markets(self):
-        return list(self.markets.values())
-
-    def rotate_markets(self, current_time):
-        # Move old and current markets & balancing_markets to
-        # `past_markets` & past_balancing_markets. We use `list()` here to get a copy since we
-        # modify the market list in-place
-        self._market_rotation(current_time=current_time, markets=self.markets,
-                              past_markets=self.past_markets)
-        if self.balancing_markets is not None:
-            self._market_rotation(current_time=current_time, markets=self.balancing_markets,
-                                  past_markets=self.past_balancing_markets)
-        self._indexed_future_markets = {
-            m.id: m for m in self.all_future_spot_markets
-        }
-
-    def _market_rotation(self, current_time, markets, past_markets):
-        for timeframe in list(markets.keys()):
-            if timeframe < current_time:
-                market = markets.pop(timeframe)
-                market.readonly = True
-                self._delete_past_markets(past_markets)
-                past_markets[timeframe] = market
-                self.log.trace("Moving {t:%H:%M} {m} to past"
-                               .format(t=timeframe, m=past_markets[timeframe].name))
-
-    def _delete_past_markets(self, past_markets):
-        if not d3a.constants.RETAIN_PAST_MARKET_STRATEGIES_STATE:
-            delete_markets = [pm for pm in past_markets if
-                              pm not in self.markets.values()]
-            for pm in delete_markets:
-                if ConstSettings.GeneralSettings.EVENT_DISPATCHING_VIA_REDIS:
-                    past_markets[pm].redis_api.stop()
-                del past_markets[pm].offers
-                del past_markets[pm].trades
-                del past_markets[pm].offer_history
-                del past_markets[pm].notification_listeners
-                del past_markets[pm].bids
-                del past_markets[pm].bid_history
-                del past_markets[pm].traded_energy
-                del past_markets[pm]
+        # Children trade in `markets`
+        self.markets:  Dict[DateTime, Market] = OrderedDict()
+        self.balancing_markets:  Dict[DateTime, BalancingMarket] = OrderedDict()
+        self.settlement_markets: Dict[DateTime, Market] = OrderedDict()
```

Comment:
```suggestion
        self.settlement_markets: Dict[DateTime, TwoSidedMarket] = OrderedDict()
```
Since settlement_markets to be double sided